moda-contracts/src/interfaces/ICatalog.sol

```diff
@@ -0,0 +1,336 @@
+
+    /**
+     * @dev Sets the track uri
+     * @notice Only the track owner can call this function
+     * @param trackId - The id of the track
+     * @param newTrackRegistrationHash - The new registration hash of the track
+     * i.e. the updated IPFS hash of the track metadata
+     */
+    function setTrackMetadata(
+        string calldata trackId,
+        string calldata newTrackRegistrationHash
+    ) external;
+
+    /**
+     * @dev Registers a Releases contract, a release can only be created from registered
+     * Releases contract and only tokens from registered Releases contract can be
+     * listed on the marketplace.
+     * @notice Only the factory can call this function
+     * @param releases - The address of the Releases contract
+     * @param releasesOwner - The owner of the Releases contract
+     */
+    function registerReleasesContract(address releases, address releasesOwner) external;
+
+    /**
+     * @dev Unregisters a Releases Contract. This will make the Releases contract
+     * unable to create releases.
+     * @param releases - The address of the Releases contract
+     */
+    function unregisterReleasesContract(address releases) external;
+
+    /**
+     * @dev Returns the owner of a Releases contract
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesOwner(address releases) external view returns (address owner);
+
+    /**
+     * @dev Add or revoke permission for a Releases contract to release a track
+     * @param trackId - The track id of the track
+     * @param releases - The address of the Releases contract
+     * @param hasPermission - The permission to set true/false
+     */
+    function grantReleasesAccessSingle(
+        string calldata trackId,
+        address releases,
+        bool hasPermission
+    ) external;
+
+    /**
+     * @dev Sets permission for a Releases contract to be able to release
+     * all tracks owned by the caller
+     * @param artist - The address of the artist
+     * @param releases - The address of the Releases contract
+     * @param hasPermission - The permission to set true/false
+     */
+    function grantReleasesAccessAll(address artist, address releases, bool hasPermission) external;
+
+    /**
+     * @dev Returns true/false if a Releases contract has permission
+     * to release a track
+     * @param trackId - The track id of the track
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesAccessSingle(
+        string calldata trackId,
+        address releases
+    ) external view returns (bool);
+
+    /**
+     * @dev Returns true/false if a Releases contract has permission
+     * to release all tracks owned by the caller
+     * @param artist - The address of the artist
+     * @param releases - The address of the Releases contract
+     */
+    function getReleasesAccessAll(address artist, address releases) external view returns (bool);
```

Comment:
How about `getReleasesApprovalForAll`? It sounds like the ERC721 / 1155 `getApprovalForAll()`